app/forms/flood_risk_engine/steps/correspondence_contact_email_form.rb

```diff
@@ -45,29 +45,48 @@ def self.params_key
       # The helpers ; allow_blank: true, allow_nil: true  ; only seem to apply to the email_address field
       # not the confirmation, so this format prevents both format & blank messages when conf = blank
       #
-      validation :passwords_match, if: :confirmation_present? do
+      validation :passwords_match?, if: :confirmation_present? do
         # Although not specified, Rails automatically validates against a field called email_address_confirmation
         validates :email_address, confirmation: {
           message: I18n.t("#{CorrespondenceContactEmailForm.locale_key}.errors.email_address_confirmation.format")
         }
       end
-      #       validates :email_address_confirmation,
-      #                 presence: {
-      #                   message: validation_message_when("email_address_confirmation.blank"),
-      #                   unless: ->(form) { form.errors.any? }
-      #                 }
-      #
-      #       validates :email_address,
-      #                 confirmation: {
-      #                   message: validation_message_when("email_address_confirmation.format"),
-      #                   unless: ->(form) { form.errors.any? }
-      #                 }
+
       def save
         super
         enrollment.correspondence_contact ||= model
         enrollment.save
       end
 
+      # Over ride the reader used in presentation to populate the confirmation field when :
+      #   User returns after successful submission - via back or review - because we don't want to force them
+      #   to re-enter and re-confirm an already saved email address.
+      #
+      # N.B This reader used by validations as well as view so care needed as to when to populate field
+      #
+      def email_address_confirmation
+        if any_email_field_changed? && email_present? && no_email_errors?
+          enrollment.correspondence_contact.email_address
+        else
+          # important to use the Reform chain if we dont need the very specific to over ride
+          super
+        end
+      end
+
+      private
+
+      def any_email_field_changed?
+        (changed["email_address"] != true && changed["email_address_confirmation"] != true)
+      end
+
+      def email_present?
+        (enrollment && enrollment.correspondence_contact && !enrollment.correspondence_contact.email_address.blank?)
```

Comment:
[`.present?`](http://api.rubyonrails.org/classes/Object.html#method-i-present-3F) is an alternative to not `.blank?`